mmdet/datasets/pipelines/loading.py

```diff
@@ -603,30 +603,26 @@ def __call__(self, results):
             assert 'gt_bboxes' in results
             gt_bboxes = results['gt_bboxes']
             instance_num = gt_bboxes.shape[0]
+            keep = np.ones(gt_bboxes.shape[0], dtype=bool)
         if self.by_mask:
             assert 'gt_masks' in results
             gt_masks = results['gt_masks']
             instance_num = len(gt_masks)
+            keep = np.ones(len(gt_masks), dtype=bool)
```

Comment:
Seems that `instance_num` and `keep` will over-write if self.by_box and self.by_mask is True.

Reply:
len(gt_masks) -> instance_num